dask_planner/src/parser.rs

```diff
@@ -255,37 +294,43 @@ impl<'a> DaskParser<'a> {
 
     /// Parse a SQL CREATE statement
     pub fn parse_create(&mut self) -> Result<DaskStatement, ParserError> {
-        let if_not_exists =
-            self.parser
-                .parse_keywords(&[Keyword::IF, Keyword::NOT, Keyword::EXISTS]);
         let or_replace = self.parser.parse_keywords(&[Keyword::OR, Keyword::REPLACE]);
         match self.parser.peek_token() {
             Token::Word(w) => {
                 match w.value.to_lowercase().as_str() {
                     "model" => {
                         // move one token forward
                         self.parser.next_token();
+
+                        let if_not_exists = self.parser.parse_keywords(&[
+                            Keyword::IF,
+                            Keyword::NOT,
+                            Keyword::EXISTS,
+                        ]);
+
                         // use custom parsing
                         self.parse_create_model(if_not_exists, or_replace)
                     }
+                    "schema" => {
+                        // move one token forward
+                        self.parser.next_token();
+
+                        let if_not_exists = self.parser.parse_keywords(&[
+                            Keyword::IF,
+                            Keyword::NOT,
+                            Keyword::EXISTS,
+                        ]);
+
+                        // use custom parsing
+                        self.parse_create_schema(if_not_exists, or_replace)
+                    }
                     "table" => {
                         // move one token forward
                         self.parser.next_token();
                         // use custom parsing
                         self.parse_create_table(or_replace)
                     }
                     _ => {
-                        if if_not_exists {
-                            // Go back three tokens if IF NOT EXISTS was consumed
-                            self.parser.prev_token();
-                            self.parser.prev_token();
-                            self.parser.prev_token();
-                        }
-                        if or_replace {
-                            // Go back two tokens if OR REPLACE was consumed
-                            self.parser.prev_token();
-                            self.parser.prev_token();
-                        }
```

Comment:
Unless I'm missing something, don't we need to keep this block since we are parsing for `OR REPLACE` before we enter the match?

If we follow the above review comment, maybe we could move the `OR REPLACE` parsing to each match case, and wrap it up along with the `IF NOT EXISTS` parsing in a utility function.